src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs

```diff
@@ -251,6 +252,11 @@ private static BsonValue TranslateConstant(Expression node)
             return value;
         }
 
+        private BsonValue TranslateDocument(DocumentExpression node)
+        {
+            return "$$ROOT";
```

Comment:
NOTE: I would say that `$$CURRENT` _should_ suit better here than `$$ROOT`, but in our implementation `{document}` is always the whole document so so they are equivalents for us

Reply:
`$$ROOT` always points to the document currently being processed whereas `$$CURRENT` starts out pointing to `$$ROOT` but is modifiable. Since our aggregations don't explicitly modify the value of `$$CURRENT`, `$$CURRENT` will always point to the same place as `$$ROOT`. See [Variables in Aggregation Expressions](https://docs.mongodb.com/manual/reference/aggregation-variables/) for more info.

Reply:
Searching the source code I find that `new DocumentExpression` occurs in many places.

Why do we think that EVERY `DocumentExpression` should be translated to `$$ROOT`.

Are tests passing simply because we don't have the right tests?

Reply:
You stated in triage (or maybe on Slack) that you have studied this question and I understand what you said as:

1. while it is true that `DocumentExpression` is used in many places
2. when execution reaches THIS place you have verified that the `DocumentExpression` ALWAYS refers to `$$ROOT`

I will trust you on this one...